### PR TITLE
CREATE DATABASE [IF NOT EXISTS] ...

### PR DESCRIPTION
**Pull request by otoolep**
Add support for optional `IF NOT EXISTS` to `CREATE DATABASE` command. This is generally useful to ensure a database exists with a default retention policy, without having to bother with error checking.

It is also specifically useful for the upcoming monitoring functionality.

This was also requested in https://github.com/influxdb/influxdb/issues/2458 and follows the pattern of mysql and sqlite.

### CODE REVIEW

**Comment by otoolep** [comment]
@dgnorton @beckettsean

**Comment by beckettsean** [comment]
works for me. I think we need to add `NOT` to https://github.com/influxdb/influxdb/blob/master/influxql/INFLUXQL.md#keywords and to https://github.com/influxdb/influxdb.com/blob/master/content/docs/v0.9/query_language/spec.md#keywords, which is basically a copy of the former in the Docs repo.

Anyone have a better way to keep those two in sync than just manually copying?

**Comment by otoolep** [comment]
@beckettsean -- I don' thave a better way, but I will update `INFLUXQL.md` as part of this PR.

**Comment by otoolep** [comment]
@beckettsean -- we also need to update the documents around `CREATE DATABASE` too, of course.

**Comment by otoolep** [comment]
Branch is green on Circle, as can be seen here https://circleci.com/gh/influxdb/influxdb/tree/if_not_exists

Final commit was docs only, so CI was skipped.

**Comment by otoolep** [comment]
Added another test, so Circle should kick off again.

File: influxql/parser.go [hunk 1]
```diff
@@ -1250,6 +1250,19 @@ func (p *Parser) parseCreateContinuousQueryStatement() (*CreateContinuousQuerySt
...
 	stmt := &CreateDatabaseStatement{}
 
+	// Look for "IF NOT EXISTS"
+	if tok, _, _ := p.scanIgnoreWhitespace(); tok == IF {
+		if tok, pos, lit := p.scanIgnoreWhitespace(); tok != NOT {
+			return nil, newParseError(tokstr(tok, lit), []string{"NOT"}, pos)
+		}
+		if tok, pos, lit := p.scanIgnoreWhitespace(); tok != EXISTS {
+			return nil, newParseError(tokstr(tok, lit), []string{"EXISTS"}, pos)
+		}
```
Comment (dgnorton) [inline influxql/parser.go]: `NOT EXISTS` parsing can be shortened to this...
```
if err := p.parseTokens([]Token{NOT, EXISTS}); err != nil {
   return nil, err
}
```

**Comment by dgnorton** [comment]
@otoolep  one minor thing but otherwise +1

@beckettsean yes, keeping those two in sync is something that needs to be automated.  I'll look into it.

**Comment by otoolep** [comment]
Thanks for the pointer @dgnorton

**Pull request merged**